### PR TITLE
[python] Additional experiment-query unit tests

### PR DESCRIPTION
**Pull request by bkmartinjr**
## Issue and/or context:

#778  - need to increase unit tests

## Changes:

Added additional unit tests for `somacore.ExperimentAxisQuery` and related.

## Notes for Reviewer:

Note: routines in `_fast_csr` compiled by numba are tested, but codecov will not report them as such

### CODE REVIEW

File: apis/python/tests/test_experiment_query.py [hunk 1]
```diff
@@ -335,24 +416,44 @@ def test_error_corners(soma_experiment: soma.Experiment):
     """Verify a couple of error conditions / corner cases."""
     assert soma_experiment.exists()
 
+    # Unkonwn Measurement name
```
Comment (johnkerl) [inline apis/python/tests/test_experiment_query.py:419]: Unknown measurement name

**Review by johnkerl — APPROVED** [review]
🚢
